workers/data_refinery_workers/processors/salmon.py

```diff
@@ -123,55 +140,19 @@ def _download_index(job_context: Dict) -> Dict:
     read length from Permanent Storage.
     """
-    batch = job_context["batches"][0]
-    try:
-        index_batch = BatchKeyValue.objects.select_related("batch").filter(
-            batch__source_type=Downloaders.TRANSCRIPTOME_INDEX.value,
-            batch__organism_id=batch.organism_id,
-            batch__status=BatchStatuses.PROCESSED.value,
-            key="kmer_size",
-            value=job_context["kmer_size"]
-        ).all()[0].batch
-
-        index_file = index_batch.files[0]
-    except:
-        logger.exception("Failed to find an index for organism %s with kmer_size of %s.",
-                         batch.organism_name,
-                         job_context["kmer_size"],
-                         processor_job=job_context["job_id"],
-                         batch=batch.id)
-
-        failure_template = "Failed to find an index for organism {} with kmer_size of {}."
-        job_context["job"].failure_reason = failure_template.format(batch.organism_name,
-                                                                    job_context["kmer_size"])
-        job_context["success"] = False
-        return job_context
-
-    temp_dir = batch.files[0].get_temp_dir(job_context["job_dir_prefix"])
-    job_context["index_directory"] = os.path.join(temp_dir, "index")
-    os.makedirs(job_context["index_directory"], exist_ok=True)
-    try:
-        # The index_batch file will be downloaded to a directory
-        # specific to that batch.
-        temp_path = index_file.download_processed_file(job_context["job_dir_prefix"])
-        with tarfile.open(temp_path, "r:gz") as tarball:
-            tarball.extractall(temp_dir)
-    except:
-        logger.exception("Failed to download and extract index tarball %s",
-                         index_file.name,
-                         index_batch=index_batch.id,
-                         index_file=index_file.id,
-                         processor_job=job_context["job_id"],
-                         batch=batch.id)
-
-        failure_template = "Failed to download and extract index tarball {}"
-        job_context["job"].failure_reason = failure_template.format(index_file.name)
-        job_context["success"] = False
-        return job_context
-    finally:
-        # Remove temp dir for the index batch since we've extracted it to
-        # the temp dir for this job.
-        index_file.remove_temp_directory(job_context["job_dir_prefix"])
+    logger.info("Downloading and installing index..")
+
+    index_object = OrganismIndex.objects.filter(organism=job_context['organism'], index_type="TRANSCRIPTOME_" + job_context["index_length"].upper()).order_by('created_at')[0]
+    result = index_object.result
+    files = ComputedFile.objects.filter(result=result)
+    job_context["index_unpacked"] = '/'.join(files[0].absolute_file_path.split('/')[:-1]) + '/index'
+    job_context["index_directory"] = job_context["index_unpacked"] + "/index"
```

Comment:
Can you make this file path stuff a bit more clear? It's fairly confusing especially because it looks like a `/index/index` directory will be made and it's not obvious why.

Reply:
I removed the extra `/index`, but I'm a little worried it was in there for a reason I don't remember now. I think it should be fine without it but I'm gonna run some experiments to make sure.